core/src/main/java/org/apache/accumulo/core/spi/security/Auth.java

```diff
@@ -0,0 +1,74 @@
+package org.apache.accumulo.core.spi.security;
+
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.security.tokens.AuthenticationToken;
+import org.apache.accumulo.core.security.Authorizations;
+
+/**
+ * Pluggable authentication and authorization module returned by {@link SecurityModule#auth()}.
+ */
+public interface Auth {
+
+  /**
+   * Verify the userPrincipal and serialized {@link AuthenticationToken} are valid.
+   *
+   * @param userPrincipal
+   *          the user to authenticate
+   * @param token
+   *          the {@link AuthenticationToken}
+   * @return boolean true if successful or false otherwise
+   * @throws AccumuloSecurityException
+   *           if a problem occurred during authentication
+   */
+  boolean authenticate(String userPrincipal, AuthenticationToken token)
```

Comment:
I think it'd be useful to return an AuthenticatedUser object instead of a boolean.